### PR TITLE
[User management] Use collate "C" for Postgres and add test for collation for all backends. [DPP-908]

### PR DESCRIPTION
**Pull request by pbatko-da**
In Oracle it should pass as our tests use an image that has NLS_LANG='American' which implies NLS_SORT='BINARY'

### CODE REVIEW

**Comment by cocreature** [comment]
CI failure looks fairly legit
```
UserManagementServiceIT
- [UserManagementServiceIT:TestListUsersRequestPageSizeZero] Exercise ListUsers rpc: Requesting page of size zero means requesting server's default page size, which is larger than zero ... Assertion failed at line 487
  two objects are supposed to be equal but they are not
  values are not the same
  => Obtained
  User(
    id = "3ac4b99f-270f-446e-aff5-6bdac2768114",
    primaryParty = ""
  )
  => Diff (- obtained, + expected)
   User(
  -  id = "3ac4b99f-270f-446e-aff5-6bdac2768114",
  +  id = "participant_admin",
     primaryParty = ""
  com.daml.ledger.api.testtool.infrastructure.AssertionErrorWithPreformattedMessage: two objects are supposed to be equal but they are not
  	at com.daml.ledger.api.testtool.infrastructure.Assertions$.assertEquals(Assertions.scala:55)
  	at com.daml.ledger.api.testtool.suites.v1_8.UserManagementServiceIT.$anonfun$new$81(UserManagementServiceIT.scala:487)
  	at com.daml.ledger.api.testtool.suites.v1_8.UserManagementServiceIT.$anonfun$new$81$adapted(UserManagementServiceIT.scala:479)
  	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:467)
  	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
  	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
  	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
  	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
  	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
  	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by pbatko-da** [comment]
Thanks @cocreature 
Setting that test case to run sequentially is it was likely a race between the the test cases

File: ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V118__add_user_managment.sql [hunk 1]
```diff
@@ -5,22 +5,18 @@
...
     created_at          BIGINT          NOT NULL
 );
-COMMENT ON COLUMN participant_users.created_at IS
-    'Up to and including Postgres v13 EXTRACT function returns double type and thus this column is susceptible to microseconds rounding errors from around year 2250. In contrast, in Postgres v14 EXTRACT returns numeric type which effectively has arbitrary precision for this use case';
 
 CREATE TABLE participant_user_rights (
     user_internal_id    INTEGER         NOT NULL REFERENCES participant_users (internal_id) ON DELETE CASCADE,
     user_right          INTEGER         NOT NULL,
-    for_party           VARCHAR(512),
+    for_party           VARCHAR(512)    COLLATE "C",
```
Comment (nmarton-da) [inline ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V118__add_user_managment.sql:16]: Why do we need this?

File: ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V118__add_user_managment.sql [hunk 2]
```diff
@@ -5,22 +5,18 @@
 
 CREATE TABLE participant_users (
     internal_id         SERIAL          PRIMARY KEY,
-    user_id             VARCHAR(256)    NOT NULL UNIQUE,
-    primary_party       VARCHAR(512),
+    user_id             VARCHAR(256)    NOT NULL UNIQUE COLLATE "C",
+    primary_party       VARCHAR(512)    COLLATE "C",
```
Comment (nmarton-da) [inline ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V118__add_user_managment.sql:9]: Why do we need this?

File: ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V118__add_user_managment.sql [hunk 3]
```diff
@@ -5,22 +5,18 @@
```
Comment (nmarton-da) [inline ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V118__add_user_managment.sql:5]: normally we add migrations, in this case it's okay just to change it

**Review by nmarton-da — APPROVED** [review]

File: ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V118__add_user_managment.sql (hunk 2 above)
Comment (pbatko-da) [reply ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V118__add_user_managment.sql]: Discussed offline, removed

**Pull request merged**